c++/src/kj/async-inl.h

```diff
@@ -2326,7 +2378,8 @@ class Coroutine<T>::Awaiter: public AwaiterBase {
   // awaited promise result.
 
 public:
-  explicit Awaiter(Promise<U> promise): AwaiterBase(PromiseNode::from(kj::mv(promise))) {}
+  explicit Awaiter(Promise<U>&& promise): AwaiterBase(PromiseNode::from(kj::mv(promise))) {}
+  explicit Awaiter(BranchForCoAwaitPromise<U>& branch): AwaiterBase(branch.node()) {}
```

Comment:
It might be reasonable to move the calls to `PromiseNode::from()` and `branch.node()` into `await_transform()`, and have `Awaiter` accept only `OwnPromiseNode`, so that adding new awaitables like this doesn't require modifying both Coroutine and Coroutine::Adapter.